src/utils/private.ts

```diff
@@ -0,0 +1,49 @@
+import _sodium from "libsodium-wrappers";
+import { getBotConfig, getBotContext } from "../bindings";
+import { Payload } from "../types";
+
+const CONFIG_REPO = "ubiquibot-config";
+const KEY_PATH = ".github/ubiquibot-config.yml";
+const KEY_NAME = "PSK";
+const KEY_PREFIX = "HSK_";
+
+export const getPrivateKey = async (): Promise<string | undefined> => {
+  try {
+    await _sodium.ready;
+    const sodium = _sodium;
+    const context = getBotContext();
+    const {
+      sodium: { publicKey, privateKey },
+    } = getBotConfig();
+    const payload = context.payload as Payload;
+
+    if (publicKey === "" || privateKey === "") {
+      return undefined;
+    }
+
+    const { data } = await context.octokit.rest.repos.getContent({
+      owner: payload.organization?.login!,
+      repo: CONFIG_REPO,
+      path: KEY_PATH,
+      mediaType: {
+        format: "raw",
+      },
+    });
+
+    if (data) {
+      const cipherText = (data as any).startsWith(KEY_NAME) ? (data as any).replace(`${KEY_NAME}: `, "").replace(`"`, "") : undefined;
```

Comment:
Isn't it better to use a [yaml parser](https://www.npmjs.com/package/yaml) here?

If there are new variables added then it is no more possible to get only a private key

Reply:
@rndquu is right. we shouldn't parse it manually. A couple of yaml parser exist as npm packages. lets use either of them.

Reply:
updated with the yml parser